### PR TITLE
Fixes to client starter

### PR DESCRIPTION
**Pull request by mlincett**
- make client args optional
- add requirements for `millipede_wilks` (AVX, CVMFS access, I3DATA env variables in Singularity)

### CODE REVIEW

File: resources/client_starter.py [hunk 1]
```diff
@@ -69,20 +69,27 @@ def make_condor_job_description(  # pylint: disable=too-many-arguments
...
     # write
     submit_dict = {
         "executable": "/bin/sh",
         "arguments": f"/usr/local/icetray/env-shell.sh python -m skymap_scanner.client {client_args} --client-startup-json ./{client_startup_json.name}",
         "+SingularityImage": singularity_image,
+        "Requirements": "HAS_CVMFS_icecube_opensciencegrid_org && has_avx",
         "getenv": "SKYSCAN_*, EWMS_*, RABBITMQ_*, PULSAR_UNACKED_MESSAGES_TIMEOUT_SEC",
         "output": str(logs_subdir / "client-$(ProcId).out"),
-        "environment": environment,
+        "environment": f'"{environment}"',
```
Comment (ric-evans) [inline resources/client_starter.py:92]: Are the quotes needed here?

File: resources/client_starter.py [hunk 2]
```diff
@@ -253,15 +260,16 @@ def wait_for_file(waitee: Path, wait_time: int) -> Path:
...
+    if args.client_args is not None:
+        for carg_value in args.client_args:
+            carg, value = carg_value.split(":", maxsplit=1)
+            client_args += f" --{carg} {value} "
+            LOGGER.info(f"Client Args: {client_args}")
+            if "--client-startup-json" in client_args:
+                raise RuntimeError(
+                    "The '--client-args' arg cannot include \"--client-startup-json\". "
+                    "This needs to be given to this script explicitly ('--client-startup-json')."
+                )
```
Comment (ric-evans) [inline resources/client_starter.py:272]: Looks like this was indented one too many levels

File: resources/client_starter.py [hunk 3]
```diff
@@ -69,20 +69,27 @@ def make_condor_job_description(  # pylint: disable=too-many-arguments
...
+
     if not os.getenv("RABBITMQ_HEARTBEAT"):
-        environment = f"RABBITMQ_HEARTBEAT=600 {environment}"
+        vars.append("RABBITMQ_HEARTBEAT=600")
     if not os.getenv("EWMS_PILOT_QUARANTINE_TIME"):
-        environment = f"EWMS_PILOT_QUARANTINE_TIME=1800 {environment}"
+        vars.append("EWMS_PILOT_QUARANTINE_TIME=1800")
+
+    for prefix in ["APPTAINERENV_", "SINGULARITYENV_"]:
+        vars.append(f"{prefix}I3_DATA=/cvmfs/icecube.opensciencegrid.org/data")
```
Comment (ric-evans) [inline resources/client_starter.py:80]: Can you add a comment explaining why this is happening, or even a URL to https://github.com/icecube/skymap_scanner/issues/135#issuecomment-1449063054?

File: resources/client_starter.py [hunk 4]
```diff
@@ -69,20 +69,27 @@ def make_condor_job_description(  # pylint: disable=too-many-arguments
...
+
+    environment = " ".join(vars)
```
Comment (ric-evans) [inline resources/client_starter.py:82]: You could use this `join()` directly in the actual dictionary assignment below. But it's not a deal breaker

**Review by ric-evans — CHANGES_REQUESTED** [review]

File: resources/client_starter.py (hunk 2 above)
Comment (mlincett) [reply resources/client_starter.py:272]: Maybe? this comes out of `black`.

Comment (ric-evans) [reply resources/client_starter.py:272]: Oh, I'm not talking about style. When it logs `client_args` & checks for `"--client-startup-json"`, this should be outside the for-loop--look at the diff.

**Review by ric-evans — CHANGES_REQUESTED** [review]

File: resources/client_starter.py (hunk 2 above)
Comment (mlincett) [reply resources/client_starter.py:272]: Ah, right. Sorry, I had tunnel-vision.

File: resources/client_starter.py (hunk 4 above)
Comment (mlincett) [reply resources/client_starter.py:82]: True, but conceptually I like the separation. If in the future we decide to build the "environment" string differently we do not have to change the code in the submit_dict block.

File: resources/client_starter.py (hunk 1 above)
Comment (ric-evans) [reply resources/client_starter.py:92]: Sure, condor does what condor wants, so who knows!

**Review by ric-evans — APPROVED** [review]
Thanks for your hard work here!

**Pull request merged**